service/src/main/java/bio/terra/cbas/util/BearerTokenFilter.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.cbas.util;
+
+import static org.springframework.web.context.request.RequestAttributes.SCOPE_REQUEST;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+/**
+ * Servlet filter that inspects the incoming request for an LeoToken=... cookie, and saves any token
+ * it finds into the current thread-local RequestContextHolder.
```

Comment:
This is probably not the right way to go any more. The `LeoToken` cookie was a bit of a hack to allow UIs to work between browser navigations to different hosts, but might disappear soon (because 1. we don't need them any more, and 2. browsers are moving away from allowing these kinds of cookies). Luckily, we should be able to assume that the requests reaching CBAS have a much more standard `Authorization` header on them, which will be of the form `Authorization: Bearer <token>`

Reply:
I was wondering about this... since we send an Authorization header from Terra-UI but a LeoToken from CBAS-UI. I will change this back to Authorization header, which will agree with WDS as well